lib/doorkeeper/oauth/token.rb

```diff
@@ -55,7 +55,11 @@ def self.from_request(request, *methods)
 
       def self.authenticate(request, *methods)
         if token = from_request(request, *methods)
-          AccessToken.by_token(token)
+          access_token = AccessToken.by_token(token)
+          if access_token && access_token.previous_refresh_token.present?
+            access_token.revoke_previous_refresh_token!
```

Comment:
Do we need this conditional and extra `revoke_previous_refresh_token` method? Can we directly always revoke it, whether it already is or not?